meeshkan/server/server/rest.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from typing import Mapping
+from openapi_typed import OpenAPIObject
+from http_types import Request, Response
+import requests
+import json
+import copy
+
+logger = logging.getLogger(__name__)
+
+
+class RestMiddlewareManager:
+    def __init__(self):
+        self._endpoints = set([])
+
+
+    def get(self):
+        return list(self._endpoints)
+    
+    def clear(self, url=None):
+        if not url:
+            self._endpoints = set([])
+        else:
+            self._endpoints.remove(url)
+
+    def add(self, url):
+        self._endpoints.add(url)
+    
+    def spew(self, request: Request, schemas: Mapping[str, OpenAPIObject]):
+        _request = copy.deepcopy(request)
+        _request['body'] = str(_request['body'])
```

Comment:
What is the reason to deepcopy here besides casting body to string? A request body should be a string initially.

Reply:
For some reason it was showing up as `bytes` in some requests. May be worth debugging...